src/webui/www/private/scripts/client.js

```diff
@@ -95,7 +95,51 @@ const getShowFiltersSidebar = function() {
     return (show === null) || (show === 'true');
 };
 
+function isValidIpv4Address(ip) {
+    // Regular expression to validate IPv4 address format
+    const ipv4Regex = /^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/;
+    return ipv4Regex.test(ip);
+}
+
+function isValidIpv6Address(ip) {
+    // Regular expression to validate IPv6 address format
+    const ipv6Regex = /^(?:[A-F0-9]{1,4}:){7}[A-F0-9]{1,4}$/i;
+    return ipv6Regex.test(ip);
+}
+
+// getHost emulate the GUI version `QString getHost(const QString &url)`
+function getHost(url) {
+    // We want the domain + tld. Subdomains should be disregarded
+    // If failed to parse the domain or IP address, original input should be returned
+
+    try {
+        const parsedUrl = new URL(url);
+        // host: "example.com:8443"
+        // hostname: "example.com"
+        const host = parsedUrl.hostname;
+        if (!host) {
+            return url;
+        }
+
+        // host is in IP format
+        if (isValidIpv4Address(host) || isValidIpv6Address(host)) {
+            return host;
+        }
+
+        return host.split(/\./).slice(-2).join('.');
+    }
+    catch (error) {
+        console.error(error);
```

Comment:
We avoid polluting the console, make sure to trim it off before merging.

Reply:
this is just a safe guard for `new URL(url)`, in general, it is not possible to trigger an error.
I got `console.error(error);` removed, is it OK now?